### PR TITLE
fix: Combobox now collapses when mousing up outside of listbox after having moused within it

### LINKED ISSUE #29629: [Bug]: Combobox - Focus can be moved outside of an expanded combobox without it collapsing
### Library

React Components / v9 (@fluentui/react-components)

### System Info

```shell
Not needed
```

### Are you reporting Accessibility issue?

no

### Reproduction

Reproduces in the [Fluent UI v9 docsite](https://react.fluentui.dev/?path=/docs/components-combobox--default)

## Actual Behavior

Mousing down over a `Combobox` option and subsequently mousing up outside of the expanded `Combobox` moves focus away from it but does not collapse it, which can cause multiple `Comboboxes` to be open at the same time.

Bug

## Expected Behavior

Mousing down over a `Combobox` option and subsequently mousing up outside of the expanded `Combobox` should collapse the `Combobox`

### Logs

_No response_

### Requested priority

Normal

### Products/sites affected

OOUI

### Are you willing to submit a PR to fix?

yes

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.

### PR DESCRIPTION
**Pull request by khmakoto**
## Previous Behavior

Mousing down over a `Combobox` option and subsequently mousing up outside of the expanded `Combobox` moved focus away from it but did not collapse it, which could cause multiple `Comboboxes` to be open at the same time.

Bug

## New Behavior

Mousing down over a `Combobox` option and subsequently mousing up outside of the expanded `Combobox` now collapses the `Combobox`.

Combobox

## Related Issue(s)

- Fixes #29629

### CODE REVIEW

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29682/perf-test-react-components/Avatar-mount_base.html">633</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29682/perf-test-react-components/Avatar-mount.html">625</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29682/perf-test-react-components/Button-mount_base.html">310</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29682/perf-test-react-components/Button-mount.html">301</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Field</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29682/perf-test-react-components/Field-mount_base.html">1101</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29682/perf-test-react-components/Field-mount.html">1095</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29682/perf-test-react-components/FluentProvider-mount_base.html">691</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29682/perf-test-react-components/FluentProvider-mount.html">714</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29682/perf-test-react-components/FluentProviderWithTheme-mount_base.html">77</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29682/perf-test-react-components/FluentProviderWithTheme-mount.html">76</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29682/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">63</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29682/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">65</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29682/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">82</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29682/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">72</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29682/perf-test-react-components/MakeStyles-mount_base.html">862</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29682/perf-test-react-components/MakeStyles-mount.html">864</a></td>
            <td>50000</td>
            <td></td>
           </tr>
<tr>
            <td>Persona</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29682/perf-test-react-components/Persona-mount_base.html">1730</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29682/perf-test-react-components/Persona-mount.html">1713</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29682/perf-test-react-components/SpinButton-mount_base.html">1351</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29682/perf-test-react-components/SpinButton-mount.html">1345</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

**Comment by fabricteam** [comment]
## 📊 Bundle size report

| Package & Exports                                                                                                             |     Baseline (minified/GZIP) |                           PR |                                                                                                                                                                                                  Change |
| :---------------------------------------------------------------------------------------------------------------------------- | ---------------------------: | ---------------------------: | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------: |
| <samp>react-combobox</samp> <br /> <abbr title='bundle-size/Combobox.fixture.js'>Combobox (including child components)</abbr> | `91.781 kB`<br />`29.935 kB` | `92.005 kB`<br />`29.995 kB` | `224 B` <br />`60 B`  |
| <samp>react-combobox</samp> <br /> <abbr title='bundle-size/Dropdown.fixture.js'>Dropdown (including child components)</abbr> | `90.223 kB`<br />`29.601 kB` | `90.447 kB`<br />`29.655 kB` | `224 B` <br />`54 B`  |

<details>
<summary>Unchanged fixtures</summary>

| Package & Exports                                                                                                                                                             |          Size (minified/GZIP) |
| ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ----------------------------: |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ButtonProviderAndTheme.fixture.js'>react-components: Button, FluentProvider & webLightTheme</abbr>              |  `70.002 kB`<br />`20.161 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/MultipleComponents.fixture.js'>react-components: Accordion, Button, FluentProvider, Image, Menu, Popover</abbr> | `207.417 kB`<br />`59.289 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ProviderAndTheme.fixture.js'>react-components: FluentProvider & webLightTheme</abbr>                            |      `42.286 kB`<br />`14 kB` |
| <samp>react-portal-compat</samp> <br /> <abbr title='bundle-size/PortalCompat.fixture.js'>PortalCompatProvider</abbr>                                                         |    `6.651 kB`<br />`2.252 kB` |

</details>
<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/a0bd42391c0a259558383e0ea6077617485aa234'>a0bd42391c0a259558383e0ea6077617485aa234</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## 🕵 fluentuiv9 No visual regressions between this PR and main
<div id="vrtCommentfluent-ui_VRT_Pipeline_v9"/>

File: packages/react-components/react-combobox/src/utils/useTriggerListboxSlots.ts [hunk 1]
```diff
@@ -91,9 +94,17 @@ export function useTriggerListboxSlots(
     }, listbox?.onMouseOver),
   );
 
+  const documentOnMouseUp = useEventCallback((ev: MouseEvent) => {
+    if (!listboxRef.current?.contains(ev.target as HTMLElement)) {
+      setOpen(ev as unknown as React.MouseEvent<HTMLElement>, false);
+    }
+    document.removeEventListener('mouseup', documentOnMouseUp);
```
Comment (smhigley) [inline packages/react-components/react-combobox/src/utils/useTriggerListboxSlots.ts:101]: I think we should either use `targetDocument` from `useFluent` / the FluentProvider, or `listboxRef.ownerDocument`

Comment (khmakoto) [reply packages/react-components/react-combobox/src/utils/useTriggerListboxSlots.ts]: You're right, I'm getting it from `useFluent` now 😄

**Review by smhigley — APPROVED** [review]

**Pull request merged**